contracts/pods/PodAdminGateway.sol

```diff
@@ -242,60 +179,61 @@ contract PodAdminGateway is CoreRef, IPodAdminGateway {
     }
 
     /// @notice Admin functionality to turn off pod membership transfer
-    /// @dev Permissioned to GOVERNOR, POD_ADMIN, GUARDIAN and the specific role
+    /// @dev Permissioned to GOVERNOR, POD_ADMIN and the specific pod admin role
     function lockMembershipTransfers(uint256 _podId)
         external
         override
-        hasAnyOfFourRoles(
+        hasAnyOfThreeRoles(
             TribeRoles.GOVERNOR,
             TribeRoles.POD_ADMIN,
-            TribeRoles.GUARDIAN,
-            getSetMembershipTransferLockRole(_podId)
+            getSpecificPodAdminRole(_podId)
         )
     {
         _setMembershipTransferLock(_podId, true);
     }
 
     /// @notice Admin functionality to turn on pod membership transfers
-    /// @dev Permissioned to GOVERNOR, POD_ADMIN, GUARDIAN and the specific role
+    /// @dev Permissioned to GOVERNOR, POD_ADMIN and the specific pod admin role
     function unlockMembershipTransfers(uint256 _podId)
         external
         override
-        hasAnyOfFourRoles(
+        hasAnyOfThreeRoles(
             TribeRoles.GOVERNOR,
             TribeRoles.POD_ADMIN,
-            TribeRoles.GUARDIAN,
-            getSetMembershipTransferLockRole(_podId)
+            getSpecificPodAdminRole(_podId)
         )
     {
         _setMembershipTransferLock(_podId, false);
     }
 
     /// @notice Internal method to toggle a pod membership transfer lock
     function _setMembershipTransferLock(uint256 _podId, bool _lock) internal {
-        ControllerV1 podController = podFactory.podController();
         podController.setPodTransferLock(_podId, _lock);
         emit PodMembershipTransferLock(_podId, _lock);
     }
 
     ///////////////  VETO CONTROLLER /////////////////
 
     /// @notice Allow a proposal to be vetoed in a pod timelock
-    /// @dev Permissioned to GOVERNOR, POD_ADMIN, GUARDIAN and specific POD_VETO_ROLE
-    function veto(uint256 _podId, bytes32 proposalId)
+    /// @dev Permissioned to GOVERNOR, POD_VETO_ADMIN, GUARDIAN, POD_ADMIN and the specific
+    ///      pod admin and guardian roles
+    function veto(
+        uint256 _podId,
+        address _podTimelock,
+        bytes32 _proposalId
+    )
```

Comment:
this change introduced a vulnerability. There is no check that the timelock input is the one associated with the pod

Reply:
Have added the check, the mistake came about when I tried to decouple the factory and this gateway. 

The `PodAdminGateway` is now a dependency on the factory rather than the other way around so it can have this check:

```
require(
            _podTimelock == podFactory.getPodTimelock(_podId),
            "PodId and timelock mismatch"
        );
```

The `admin` is made available to the factory when it seeks to deploy a pod, again via the `PodConfig` that is passed to `factory.createOptimisticPod()`. This also makes it easier to create new pods with an updated admin